src/rgw/CMakeLists.txt

```diff
@@ -434,49 +342,316 @@ target_include_directories(radosgw
+
+if(WITH_RADOSGW_MOTR)
+  find_package(motr REQUIRED)
+  set(rgw_sal_motr_srcs rgw_sal_motr_plugin.cc)
+  add_library(rgw_sal_motr SHARED ${rgw_sal_motr_srcs})
+  target_include_directories(rgw_sal_motr
+    PUBLIC "services"
+    PUBLIC "${CMAKE_SOURCE_DIR}/src/rgw"
+    PUBLIC "${LUA_INCLUDE_DIR}"
+    PUBLIC "/usr/include/motr")
+  target_compile_options(rgw_sal_motr PRIVATE "-Wno-attributes")
+  target_link_libraries(rgw_sal_motr PRIVATE motr::motr)
+
+if(WITH_LTTNG)
+    # rgw/rgw_op.cc includes "tracing/rgw_op.h"
+    # rgw/rgw_rados.cc includes "tracing/rgw_rados.h"
+    add_dependencies(rgw_sal_motr rgw_op-tp rgw_rados-tp)
+endif()
+
+if(WITH_JAEGER)
+    add_dependencies(rgw_sal_motr jaeger_base)
+    target_link_libraries(rgw_sal_motr PUBLIC jaeger_base)
+endif()
+
+  install(TARGETS rgw_sal_motr DESTINATION ${rgw_plugin_dir})
+endif()
+
+if(WITH_RADOSGW_DAOS)
+  find_package(DAOS REQUIRED)
+  set(rgw_sal_daos_srcs rgw_sal_daos_plugin.cc)
+  add_library(rgw_sal_daos SHARED ${rgw_sal_daos})
+  target_include_directories(rgw_sal_daos
+    PUBLIC "services"
+    PUBLIC "${CMAKE_SOURCE_DIR}/src/rgw"
+    PUBLIC "${LUA_INCLUDE_DIR}"
+    PUBLIC ${PC_DAOS_INCLUDEDIR} )
+  target_compile_options(rgw_sal_daos PRIVATE "-Wno-attributes")
+  link_directories( ${PC_DAOS_LIBRARY_DIRS} )
+  target_link_libraries(rgw_sal_daos PRIVATE daos dfs ds3 uuid duns)
+
+if(WITH_LTTNG)
+    # rgw/rgw_op.cc includes "tracing/rgw_op.h"
+    # rgw/rgw_rados.cc includes "tracing/rgw_rados.h"
+    add_dependencies(rgw_sal_daos rgw_op-tp rgw_rados-tp)
+endif()
+
+if(WITH_JAEGER)
+    add_dependencies(rgw_sal_daos jaeger_base)
+    target_link_libraries(rgw_sal_daos PUBLIC jaeger_base)
+endif()
+
+  install(TARGETS rgw_sal_daos DESTINATION ${rgw_plugin_dir})
+endif()
+
+if(WITH_RADOSGW_DBSTORE)
+  add_subdirectory(driver/dbstore)
+  set(rgw_sal_dbstore_srcs rgw_sal_dbstore_plugin.cc)
+  add_library(rgw_sal_dbstore SHARED ${rgw_sal_dbstore_srcs})
+  target_include_directories(rgw_sal_dbstore
+    PUBLIC "services"
+    PUBLIC "${CMAKE_SOURCE_DIR}/src/rgw"
+    PUBLIC "${LUA_INCLUDE_DIR}")
+
+if(WITH_LTTNG)
+    # rgw/rgw_op.cc includes "tracing/rgw_op.h"
+    # rgw/rgw_rados.cc includes "tracing/rgw_rados.h"
+    add_dependencies(rgw_sal_dbstore rgw_op-tp rgw_rados-tp)
+endif()
+
+if(WITH_JAEGER)
+    add_dependencies(rgw_sal_dbstore jaeger_base)
+    target_link_libraries(rgw_sal_dbstore PUBLIC jaeger_base)
+endif()
+
+  target_link_libraries(rgw_sal_dbstore PUBLIC dbstore)
+
+  install(TARGETS rgw_sal_dbstore DESTINATION ${rgw_plugin_dir})
+endif()
+
+set(radosgw_admin_srcs
+  rgw_admin.cc
+  rgw_orphan.cc
+  rgw_sync_checkpoint.cc)
 add_executable(radosgw-admin ${radosgw_admin_srcs})
-target_link_libraries(radosgw-admin ${rgw_libs} librados
-  cls_rgw_client cls_otp_client cls_lock_client cls_refcount_client
-  cls_log_client cls_timeindex_client
+target_link_libraries(radosgw-admin ${rgw_libs}
+  rgw_sal_rados
```

Comment:
this is probably why you're seeing teuthology failures:
> radosgw-admin: error while loading shared libraries: librgw_sal_rados.so: cannot open shared object file: No such file or directory

this is linking `radosgw-admin` directly against `librgw_sal_rados.so` instead of relying on `dlopen()`. `radosgw-admin` can't find it because it's (intentionally) not installed in the `LD_LIBRARY_PATH`

try linking `rados_sal` in instead?